coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -2476,6 +2504,13 @@ type WorkspaceProvisionJob struct {
 	IsPrebuild            bool      `json:"is_prebuild,omitempty"`
 	PrebuildClaimedByUser uuid.UUID `json:"prebuild_claimed_by,omitempty"`
 	LogLevel              string    `json:"log_level,omitempty"`
+	// RunningAgentAuthTokens is *only* used for prebuilds. We pass it down when we want to rebuild a prebuilt workspace
+	// but not generate new agent tokens. The provisionerdserver will retrieve these tokens and push them down to
+	// the provisioner (and ultimately to the `coder_agent` resource in the Terraform provider) where they will be
+	// reused. Context: the agent token is often used in immutable attributes of workspace resource (e.g. VM/container)
+	// to initialize the agent, so if that value changes it will necessitate a replacement of that resource, thus
+	// obviating the whole point of the prebuild.
+	RunningAgentAuthTokens map[uuid.UUID]string `json:"running_agent_auth_tokens"`
```

Comment:
I'm very nervous about putting secret credentials into the job input, which is returned when you query the job, making it much more likely to leak. I'd feel much better if provisionerdserver queried them from the database based on the agent IDs, rather than duplicating the creds in the database.

Reply:
done